Prevent home.arpa queries from being upstreamed. RFC 8375

rrobgill opened the pull request:
RFC8375 defines the home.arpa special use domain for local dns in a home network.
These are not globally resolvable and should not be sent upstream.

**What does this PR aim to accomplish?:**

Prevent queries for home.arpa domains being sent to upstream services.

**How does this PR accomplish the above?:**

Checks if the user has specified home.arpa as their local domain, and that we have configured dnsmasq to use a suitable local server for these queries.

Otherwise, blocks queries to this domain.

**Link documentation PRs if any are needed to support this PR:**
n/a

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [X] I have read the above and my PR is ready for review. *Check this box to confirm*

DL6ER commented on src/config/dnsmasq_config.c:
```diff
@@ -547,6 +552,14 @@ bool __attribute__((nonnull(1,3))) write_dnsmasq_config(struct config *conf, boo
+	if (domain_homearpa == false)
+	{
+		fputs("# Do not forward home.arpa domains to upstream servers\n",pihole_conf);
+		fputs("server=/home.arpa/\n\n",pihole_conf);
```
```suggestion
		fputs("local=/home.arpa/\n\n",pihole_conf);
```
I know, nit-picking here, `local` is a synonym for `server` to make configuration files clearer in this case.

DL6ER commented on src/config/dnsmasq_config.c:
```diff
@@ -547,6 +552,14 @@ bool __attribute__((nonnull(1,3))) write_dnsmasq_config(struct config *conf, boo
 
+	// Ensure that home.arpa domains (RFC 8375) are not sent upstream, unless configured by
+	// user as local domain, with server setting applied above
+	if (domain_homearpa == false)
```
```suggestion
	if (!domain_homearpa)
```
just for code style uniformity

rrobgill committed:
Prevent home.arpa queries from being upstreamed. RFC 8375

Detects if user has set up home.arpa as their domain and configured
so queries are sent to it. Otherwise blocks home.arpa domains from being
sent upstream.

Signed-off-by: Rob Gill <rrobgill@protonmail.com>
